src/implicit-function.hh

```diff
@@ -0,0 +1,320 @@
+       const segments_t& inputConf, const segments_t& inputVelocity,
+       const segments_t& outputConf, const segments_t& outputVelocity)
+      {
+	ImplicitFunction* ptr = new ImplicitFunction
+	  (robot, function, inputConf, inputVelocity, outputConf,
+           outputVelocity);
+	return ImplicitFunctionPtr_t (ptr);
+      }
+
+      /// Get function that maps input variables to output variables
+      const DifferentiableFunctionPtr_t& inputToOutput () const
+      {
+        return inputToOutput_;
+      }
+
+    protected:
+      ImplicitFunction (const DevicePtr_t& robot,
+			const DifferentiableFunctionPtr_t& function,
+			const segments_t& inputConf,
+			const segments_t& inputVelocity,
+                        const segments_t& outputConf,
+			const segments_t& outputVelocity)
+	: DifferentiableFunction (robot->configSize (), robot->numberDof (),
+				  LiegroupSpace::Rn
+                                  (function->outputSpace ()->nv ())),
+	  robot_ (robot), inputToOutput_ (function),
+          inputConfIntervals_ (inputConf),
+	  inputDerivIntervals_ (inputVelocity),
+          outputConfIntervals_ (outputConf),
+	  outputDerivIntervals_ (outputVelocity), outJacobian_ (),
+          inJacobian_ (), f_qIn_ (function->outputSpace ()),
+          qOut_ (function->outputSpace ()), result_ (outputSpace ())
+      {
+	// Check input consistency
+	// Each configuration variable is either input or output
+	assert (function->inputSize () + function->outputSize () <=
+		robot->configSize ());
+	// Each velocity variable is either input or output
+	assert (function->inputDerivativeSize () +
+		function->outputDerivativeSize () <= robot->numberDof ());
+	qIn_.resize (function->inputSize ());
+	Jf_.resize (function->outputDerivativeSize (),
+                    function->inputDerivativeSize ());
+	size_type size = 0;
+	// Sum of configuration output interval sizes equal function output size
+	for (segments_t::const_iterator it = outputConf.begin ();
+	     it != outputConf.end (); ++it) {
+	  size += it->second;
+	}
+	assert (size == function->outputSize ());
+	// Sum of velocity output interval sizes equal function output
+	// derivative size
+	size = 0;
+	for (segments_t::const_iterator it = outputVelocity.begin ();
+	     it != outputVelocity.end (); ++it) {
+	  size += it->second;
+	}
+	assert (size == function->outputDerivativeSize ());
+        computeJacobianBlocks ();
+      }
+
+      /// Compute q_{output} - f (q_{input})
+      void impl_compute (LiegroupElement& result, vectorIn_t argument) const
+      {
+        hppDout (info, "argument=" << argument.transpose ());
+        // Store q_{output} in result
+	size_type index = 0;
+	for (segments_t::const_iterator it = outputConfIntervals_.begin ();
+	     it != outputConfIntervals_.end (); ++it) {
+	  qOut_.vector ().segment (index, it->second) =
+	    argument.segment (it->first, it->second);
+	  index += it->second;
+	}
```

Comment:
To keep track of a possible issue: did you use a for loop because `MatrixBlocks` does not work with `Eigen::Ref` matrices ?

Reply:
I will modify and try.